solace_prometheus_exporter.go

```diff
@@ -2865,7 +2993,12 @@ func main() {
 	}
 
 	http.HandleFunc("/solace", func(w http.ResponseWriter, r *http.Request) {
-		var err = r.ParseForm()
+		err := validateAuthentication(conf, w, r)
+		if err != nil {
+			return
```

Comment:
better do here:
		w.Header().Set("Content-Type", "text/plain; charset=utf-8") // normal header
		w.WriteHeader(http.StatusUnauthorized)

		io.WriteString(w, err + "\n")
                return


To return the error message to the requestor.